hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/EightToSevenDowngradeHandler.java

```diff
@@ -92,4 +115,52 @@ private static void downgradePartitionFields(HoodieWriteConfig config, HoodieEng
       tablePropsToAdd.put(HoodieTableConfig.PARTITION_FIELDS, tableConfig.getPartitionFieldProp());
     }
   }
+
+  static void downgradeMetadataPartitions(HoodieEngineContext context,
+                                          HoodieStorage hoodieStorage,
+                                          HoodieTableMetaClient mdtMetaClient,
+                                          Map<ConfigProperty, String> tablePropsToAdd) {
+    // Fetch metadata partition paths.
+    List<String> metadataPartitions = FSUtils.getAllPartitionPaths(
+        context,
+        hoodieStorage,
+        mdtMetaClient.getBasePath(),
+        true);
+
+    // Delete partitions.
+    List<String> validPartitionPaths = deleteMetadataPartition(context, mdtMetaClient, metadataPartitions);
+
+    // Clean the configuration.
+    tablePropsToAdd.put(TABLE_METADATA_PARTITIONS, String.join(",", validPartitionPaths));
+  }
+
+  static List<String> deleteMetadataPartition(HoodieEngineContext context,
+                                              HoodieTableMetaClient mdtMetaClient,
+                                              List<String> metadataPartitions) {
+    List<String> validPartitionPaths = new ArrayList<>();
+
+    for (String partitionPath : metadataPartitions) {
+      if (partitionPath.equals(FUNCTIONAL_INDEX.getPartitionPath())
+          || partitionPath.equals(SECONDARY_INDEX.getPartitionPath())
+          || partitionPath.equals(PARTITION_STATS.getPartitionPath())) {
+        HoodieTableMetadataUtil.deleteMetadataTablePartition(
+            mdtMetaClient, context, partitionPath, true);
+      } else {
+        validPartitionPaths.add(partitionPath);
+      }
+    }
+    return validPartitionPaths;
+  }
+
+  private static HoodieTableMetaClient createMetadataTableMetaClient(HoodieEngineContext context,
+                                                                     StoragePath basePath) {
+    return HoodieTableMetaClient.builder()
+        .setConf(HadoopFSUtils.getStorageConfWithCopy(
+            (Configuration) context.getStorageConf().unwrap()))
+        .setBasePath(
+            new Path(basePath.toString(),
+                HoodieTableMetaClient.METADATA_TABLE_FOLDER_PATH).toString())
```

Comment:
Avoid using Hadoop's `Path`
```suggestion
        .setBasePath(HoodieTableMetadata.getMetadataTableBasePath(basePath).toString())
```

Reply:
Done.